packages/paste-core/components/combobox/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {useUID} from 'react-uid';
+import {useComboboxPrimitive} from '@twilio-paste/combobox-primitive';
+import {ChevronDownIcon} from '@twilio-paste/icons/esm/ChevronDownIcon';
+import {Box} from '@twilio-paste/box';
+import {FormHelpText, FormInput, FormLabel} from '@twilio-paste/form';
+import {Text} from '@twilio-paste/text';
+
+export interface ComboboxProps {
+  autocomplete?: boolean;
+  disabled?: boolean;
+  hasError?: boolean;
+  helpText?: string | React.ReactNode;
+  id?: string;
+  items: string[];
+  labelText: string | NonNullable<React.ReactNode>;
+  name?: string;
+  placeholder?: string;
+  required?: boolean;
+}
+
+const ComboboxIconWrapper: React.FC<{children: React.ReactNode}> = ({children}) => (
+  <Box
+    alignItems="center"
+    display="inline-flex"
+    position="absolute"
+    pointerEvents="none"
+    right="space40"
+    top="50%"
+    transform="translateY(-50%)"
+    zIndex="zIndex10"
+  >
+    {children}
+  </Box>
+);
+
+/* eslint-disable react/no-array-index-key */
+const Combobox: React.FC<ComboboxProps> = ({
+  autocomplete,
+  disabled,
+  helpText,
+  hasError,
+  id,
+  items,
+  labelText,
+  name,
+  placeholder,
+  required,
+}) => {
+  const [inputItems, setInputItems] = React.useState(items);
+  const {
+    getComboboxProps,
+    getInputProps,
+    getItemProps,
+    getLabelProps,
+    getMenuProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    isOpen,
+  } = useComboboxPrimitive({
+    items: inputItems,
+    onInputValueChange: ({inputValue}) => {
+      if (autocomplete && inputValue) {
+        setInputItems(items.filter(item => item.toLowerCase().startsWith(inputValue.toLowerCase())));
+      }
+      return inputValue;
+    },
+  });
+  const helpTextId = useUID();
+  return (
+    <>
+      <FormLabel disabled={disabled} htmlFor={id} required={required} {...getLabelProps()}>
```

Comment:
Lets pull label from Combobox, and rely on composition. If you want FormLabel you can get it from `form`. Reason being this is just starting to do too much. If I just want a simple search combobox, I want to provide a hidden label, and then we have to start extending all the props to handle those edge case. 

Lets keen this simple.

Reply:
So in pulling the label, we lose the `getLabelProps` function in the hook, and get dinged with some accessibility errors.

Reply:
OK, that's fair. If you do want something to look slightly different that's what the primitive is for, for building your own.

So that leads to be the next part then. Can you create components for each of the pieces that we need to compose a new combobox with? Say Everything we use Box for.

So when I do need use the primitive, say I want to swap the input, I still might want the input wrapper, the listbox and it's options. So I could do 

```
import {FormHelpText, FormInput, FormLabel} from '@twilio-paste/form';
import {ComboboxInputWrapper, ComboboxListBox, ComboboxListboxOption} from '@twilio-paste/combobox;

<FormLabel disabled={disabled} htmlFor={id} required={required} {...getLabelProps()}>
  {labelText}
</FormLabel>
<ComboboxInputWrapper {...getComboboxProps({role: 'combobox'})}>
  <FormInput
    {...getInputProps()}
    {...getToggleButtonProps({tabIndex: 0})}
    disabled={disabled}
    hasError={hasError}
    id={id}
    name={name}
    placeholder={placeholder}
    required={required}
    type="text"
  />
  <ComboboxIconWrapper>
    <ChevronDownIcon size="sizeIcon30" decorative aria-hidden="true" />
  </ComboboxIconWrapper>
</ComboboxInputWrapper>
<ComboboxListbox
  {...getMenuProps()}
>
  {isOpen &&
    inputItems.map((item, index) => (
      <ComboboxListboxOption
        {...getItemProps({item, index})}
      >
        <Text as="span">{item}</Text>
      </ComboboxListboxOption>
    ))}
</ComboboxListbox>
{helpText && (
  <FormHelpText id={helpTextId} variant={hasError ? 'error' : undefined}>
    {helpText}
  </FormHelpText>
)}

```

Reply:
Still need to add the actual export files, but I think I worked this out. Running into one issue with spreading the props on `ComboboxListboxOption`. If I use `safelySpreadBoxProps`, it doesn't seem to want to pass the `highlightedIndex`. It will if I just spread the props.